File: two sum/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        diff_dict = {}
        for i in range(len(nums)):
            complement = int(target) - int(nums[i])
            if complement in diff_dict:
                return [diff_dict[complement], i]
            diff_dict[nums[i]] = i
        return None

if __name__ == "__main__":
    nums = [2, 7, 11, 15]
    target = 9
    sol = Solution()
    result = sol.twoSum(nums, target)
    print(result)

File: two sum/two_sum.js
class Solution {
  twoSum(nums, target) {
    let index_dict = {};
    for (let i = 0; i < nums.length; i++) {
      let complement = target - nums[i];
      if (complement in index_dict) {
        return [index_dict[complement], i];
      }
      index_dict[nums[i]] = i;
    }
    return null;
  }
}

const nums = [2, 7, 11, 15];
const target = 9;
const solution = new Solution();
const result = solution.twoSum(nums, target)
console.log(result)